File: Niveau 4/3 – Structures de données et Balayages/10 - Augmenter la fréquentation.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

int main()
{
    int n;
    cin >> n;

    vector<pair<int, int>> arr;

    for (int i = 0; i < n; i++)
    {
        int d, f;
        cin >> d >> f;

        arr.push_back({d, f});
    }

    sort(arr.begin(), arr.end());

    int ans = 0;
    int total = 0;
    double maxMoy = 0;

    for (int i = 0; i < n; i++)
    {
        total += arr[i].second;
        double moy = total / double(i + 1);

        if (moy >= maxMoy)
        {
            maxMoy = moy;
            ans = arr[i].first;
        }
    }

    cout << ans;
}

File: Niveau 4/3 – Structures de données et Balayages/10 - Augmenter la fréquentation.py
def main():
   n = int(input())
   arr = []
   
   for i in range(n):
      d, f = map(int, input().split())      
      arr.append((d, f))
      
   arr.sort()

   ans = 0
   total = 0
   maxMoy = 0
   i = 0
   
   for e in arr:
      total += e[1]
      i += 1
      moy = total / i
      
      if moy >= maxMoy:
         maxMoy = moy
         ans = e[0]
   
   print(ans)
main()